libguard/guard_interface.cpp

```diff
@@ -317,6 +327,59 @@ void clearAll()
     }
 }
 
+void clearResolvedRecord()
```

Comment:
The logic seems to be not good fro  flash performance. , need to look better option to identify the oldest record instead of position based approch. Is this same logic is used by hostboot code?

Reply:
Not sure of hostboot code. I saw they mark every record as resolved when do clear, not sure how they are handling this. We don't have any time stamp to check that's why went with this approach. May be later can see if something better can be done here.

Reply:
Still prefer to avoid unnecessary flash right , for each delete operation. incase if this is not requirement for existing hostboot point , need to relook this flow.

Reply:
Hostboot removes the resolved/invalid guard record when on space is left in file to create a new record and writes the new record in that empty slot. So this method will not do unnecessary flash write.  Opting for this method in libguard as well.
Note: In this case the record ID will come in random order.